src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.Stream.cs

```diff
@@ -0,0 +1,375 @@
+            internal ODataUtf8JsonWriteStream(ODataUtf8JsonWriter writer)
+            {
+                this.jsonWriter = writer;
+            }
+
+            public override bool CanRead => false;
+
+            public override bool CanSeek => false;
+
+            public override bool CanWrite => true;
+
+            /// <summary>
+            /// Gets the length in bytes of the stream. This is not supported by this stream.
+            public override long Length => throw new NotSupportedException();
+
+            /// <summary>
+            /// Gets or sets the position within the stream. This is not supported by this stream.
+            /// </summary>
+            public override long Position { get => throw new NotSupportedException(); set => throw new NotSupportedException(); }
+
+            /// <summary>
+            /// Flushes any buffered data to the underlying stream synchronously.
+            /// </summary>
+            public override void Flush()
+            {
+                if (buffer != null)
+                {
+                    // If there are unprocessed bytes, encode and write them as the final block.
+                    ReadOnlySpan<byte> bytesNotProcessedFromPreviousChunk = this.buffer.AsSpan().Slice(bufferPosition - bytesNotWrittenFromPreviousChunk, bytesNotWrittenFromPreviousChunk);
+
+                    this.jsonWriter.Base64EncodeAndWriteChunk(bytesNotProcessedFromPreviousChunk, isFinalBlock: true, out bytesNotWrittenFromPreviousChunk);
+
+                    // Clear the buffer since all bytes have been written.
+                    this.buffer = null;
+                    this.bufferPosition = 0;
+                    this.bytesNotWrittenFromPreviousChunk = 0;
+                }
+
+                this.jsonWriter.Flush();
+            }
+
+            /// <summary>
+            /// Flushes any buffered data to the underlying stream asynchronously.
+            /// </summary>
+            /// <param name="cancellationToken">A A cancellation token to observe while waiting for the flush operation to complete.</param>
+            /// <returns>A task representing the asynchronous flush operation.</returns>
+            public override async Task FlushAsync(CancellationToken cancellationToken)
+            {
+                if (buffer != null)
+                {
+                    // If there are unprocessed bytes, encode and write them as the final block.
+                    ReadOnlyMemory<byte> bytesNotProcessedFromPreviousChunk = this.buffer.AsMemory().Slice(bufferPosition - bytesNotWrittenFromPreviousChunk, bytesNotWrittenFromPreviousChunk);
+
+                    this.jsonWriter.Base64EncodeAndWriteChunk(bytesNotProcessedFromPreviousChunk.Span, isFinalBlock: true, out bytesNotWrittenFromPreviousChunk);
+
+                    // Clear the buffer since all bytes have been written.
+                    this.buffer = null;

```

Comment:
Does this mean we'll allocate a new buffer the next time we write some data?

Reply:
No. Not some data. An entirely different stream or write.